agent/agent.go

```diff
@@ -105,6 +105,8 @@ type Agent struct {
 	cacheRefreshInterval time.Duration
 	clusterCache         *appstatecache.Cache
 
+	inflightMu   sync.Mutex
+	inflightLogs map[string]context.CancelFunc
```

Comment:
Two things:
- Can you add a description of what this field is used for
- AFAICT the value (cancel func) is never called. Is this intentional?